src/lib/y2storage/clients/partitions_proposal.rb

```diff
@@ -92,6 +92,10 @@ def description
         }
       end
 
+      def failed?
+        @failed
+      end
```

Comment:
Is this new method used somewhere?

Reply:
In the unit tests for sure; this is the only way that information is accessible.

Reply:
Well, if you have to extend the API of a class just in order to be able to write its unit tests, that usually means the unit tests are not properly designed. They should test the already existing API, without the need to add extra methods just to lurk into the internal status of the objects.

But let's not start a useless discussion about testing theory. The code itself looks good. So let's move on. :+1:

Reply:
Frankly, I would have exported that information from the very start; it might be of interest to the caller. So, for me this kind of thing definitely belongs to an API, no matter if the current caller(s) use it or not.

Reply:
Well, this is an `Installation::ProposalClient`, so the API is strictly (pre)defined. All proposal clients must adhere to the corresponding API - a single `#run` method that responds to `MakeProposal`, `AskUser`, `Description` and `Write`.

No room for innovation or extra stuff there, since the caller is always the proposal mechanism, which iterates through all the proposal clients using that generic API and making no difference between this proposal client and any other.